src/mixer.c

```diff
@@ -597,12 +597,14 @@ static SDL_AudioSpec *Mix_LoadMusic_RW(SDL_RWops *src, int freesrc, SDL_AudioSpe
         fragment = (MusicFragment *)SDL_malloc(sizeof(*fragment));
         if (!fragment) {
             /* Uh oh, out of memory, let's return what we have */
+            fragment = last;
             break;
         }
         fragment->data = (Uint8 *)SDL_malloc(fragment_size);
         if (!fragment->data) {
             /* Uh oh, out of memory, let's return what we have */
             SDL_free(fragment);
+            fragment = last;
```

Comment:
Haven't followed the code carefully, but: there is not a chance that
`last` and `fragment` would point to the same memory here is there?
If that ever happens, it would lead to a double-free.

Reply:
I guess it's better to replace the "fragment" with "last" at the final check?

```c
    if (count > 0 && last) {
        *audio_len = (count - 1) * fragment_size + last->size; /* <------- THIS ONE*/
        *audio_buf = (Uint8 *)SDL_malloc(*audio_len);
        if (*audio_buf) {
            Uint8 *dst = *audio_buf;
            for (fragment = first; fragment; fragment = fragment->next) {
                SDL_memcpy(dst, fragment->data, fragment->size);
                dst += fragment->size;
            }
        } else {
            SDL_OutOfMemory();
            spec = NULL;
        }
    } else {
        Mix_SetError("No audio data");
        spec = NULL;
    }
```

Reply:
Mainly, because this code block adds the size of the last fragment to finalize the data block, using the last fragment. But, because of the out-of-memory failure from above, the fragment is expected to be something existing because of the `fragment->size`. So, I guess, that's really reasonable to use the `last->size` instead of the `fragment->size` here and remove `fragment = last` lines from above.